std/hash/poseidon_gl.asm

```diff
@@ -58,129 +60,61 @@ machine PoseidonGL(LASTBLOCK, operation_id) {
     pol constant C_10 = [0xe17653a29da578a1, 0x849c2656969c4be7, 0xf061274fdb150d61, 0x40b12cbf09ef74bf, 0xfceb4fda1ded0893, 0xd1b1edf76bc09c92, 0x85df9ed2e166d64f, 0x60e75c2890d15730, 0x9c22190be7f74732, 0xb93e260cae5c170e, 0xbb93ae776bb30e3a, 0xadffe8cf28449ec6, 0x1e3b9fc625b554f4, 0x38e8ee9d71a45af8, 0xddb4ac49c9efa1da, 0xc6401fe115063f5b, 0x593664c39773012c, 0x16bf1f73fb7a9c3f, 0x45d7ac9b68f71a34, 0xc368a348e46d950f, 0xe909678474e687fe, 0x17dfc8e4f7ba8a57, 0xbdf8f242e316c4ab, 0xbba2515f22909e87, 0x4cd1f1b175100206, 0x16e50d897d4634ac, 0xf6f31db1899b12d5, 0xdebe0853b1a1d378, 0x6e7c8f1a84265034, 0xdfd1c4febcc81238, 0x0]*;
     pol constant C_11 = [0xc54302f225db2c76, 0xc0572c8c08cbbbad, 0x28b8ec3ae9c29633, 0xa637093ecb2ad631, 0xfac6ff1346a41675, 0xb65481ba645c602, 0x6604df4fee32bcb1, 0xa6217d8bf660f29c, 0x5d693c1ba3ba3621, 0xb0a7eae879ddb76d, 0x28fd3b046380f850, 0x1c4dbb1c4c27d243, 0x25d64362697828fd, 0xdd5118375bf1a9b9, 0x64bb6dec369d4418, 0x8ad97b33f1ac1455, 0x4f0a2e5fb028f2ce, 0x6d1f5a59005bec17, 0xeeb76e397069e804, 0x9ef1cd60e679f284, 0xdfe89923f6c9c2ff, 0x9001a64209f21db8, 0x9e8cd55b57637ed0, 0x7248fe7705f38e47, 0x4a20358574454ec0, 0x29bff3ecb9b7a6e3, 0xc02ac5e47312d3ca, 0xa49229d24d014343, 0x164bb2de1bbeddc8, 0xbc8dfb627fe558fc, 0x0]*;
-    
-    // State of the Poseidon permutation
-    pol commit inp[8];
-    pol commit cap[4];
+    let C = [C_0, C_1, C_2, C_3, C_4, C_5, C_6, C_7, C_8, C_9, C_10, C_11];
+
+    // State of the Poseidon permutation (8 rate elements and 4 capacity elements)
+    pol commit state[STATE_SIZE];
 
     // The initial state of the Poseidon permutation
-    // (constrained to be equal to (inp[0], inp[1], ..., inp[7], cap[0], cap[1], cap[2], cap[3])
-    // in the first row and then repeated until the end of the block)
-    pol commit input_inp[8];
-    pol commit input_cap[4];
+    // (constrained to be equal to state in the first row and then repeated until
+    // the end of the block)
+    pol commit input_state[STATE_SIZE];
 
     // Add round constants
-    pol a0 = inp[0] + C_0;
-    pol a1 = inp[1] + C_1;
-    pol a2 = inp[2] + C_2;
-    pol a3 = inp[3] + C_3;
-    pol a4 = inp[4] + C_4;
-    pol a5 = inp[5] + C_5;
-    pol a6 = inp[6] + C_6;
-    pol a7 = inp[7] + C_7;
-    pol a8 = cap[0] + C_8;
-    pol a9 = cap[1] + C_9;
-    pol a10 = cap[2] + C_10;
-    pol a11 = cap[3] + C_11;
+    pol commit a[STATE_SIZE];
```

Comment:
@chriseth I don't think we have array of intermediate polynomials, do we?

This change introduces a lot of committed polynomials where it would be better to use intermediate once instead. I think we shouldn't merger it for this reason!

(BTW, the `PoseidonBn254` is like that too, but at least it's only 3 columns per intermediate value, not 12...)